odoo/addons/base/models/ir_model.py

```diff
@@ -1259,6 +1250,45 @@ def unlink(self):
 
         return result
 
+    def _process_ondelete(self):
+        """ Process the 'ondelete' of the given selection values. """
+        for selection in self:
+            Model = self.env[selection.field_id.model]
+            field = Model._fields[selection.field_id.name]
+            if not field.store or Model._abstract:
+                continue
+
+            ondelete = (field.ondelete or {}).get(selection.value)
+            if callable(ondelete):
+                ondelete(selection._get_records())
+            elif not ondelete or ondelete == 'set null':
+                selection._get_records().write({field.name: False})
+            elif ondelete == 'set default':
+                value = field.convert_to_write(field.default(Model), Model)
+                selection._get_records().write({field.name: value})
+            else:
+                assert ondelete == 'cascade'
```

Comment:
Needs a message, otherwise Python just prints `AssertionError` and you have to dig into the traceback for information. Also I'd probably have this one be an other `elif`, and move raising a more explicit exception (e.g. ValueError or something) into a separate `else` clause.

Reply:
this is just a sanity check, technically this should never happen because all of this is checked during the `_setup_attrs` and during `_process_ondelete` every `ir.model.fields.selection` should have a valid `ondelete` policy...

that being said, I will update it to another elif + else because I also think it's cleaner, it was like this originally